test/unit/formatting_stack/linters/ns_aliases/impl.clj

```diff
@@ -0,0 +1,19 @@
+(ns unit.formatting-stack.linters.ns-aliases.impl
+  (:require
+   [clojure.test :refer [are deftest]]
+   [formatting-stack.linters.ns-aliases.impl :as sut]
+   [matcher-combinators.matchers :as matchers]
+   [matcher-combinators.test :refer [match?]]))
+
+(deftest merge-aliases
+  (are [m1 m2 expected] (match? expected
+                                (sut/merge-aliases m1 m2))
+    {}                      {}                      {}
+    {'a ['b]}               {}                      {'a ['b]}
+    {}                      {'a ['b]}               {'a ['b]}
+    {'a ['b]}               {'a ['c]}               {'a (matchers/in-any-order ['b 'c])}
+    {'a ['b]}               {'a ['b]}               {'a ['b]}
+    {'a ['b 'c] 'd ['e 'f]} {'a ['g 'h] 'd ['i 'j]} {'a (matchers/in-any-order ['b 'c 'g 'h])
```

Comment:
I guess I don't like that one can turn `{'a` into `'{a` opportunistically.

An homogeneous pattern is simpler to compare visually/cognitively, else it might not be clear that you're comparing stuff of the same kind